core/src/tangram.cpp

```diff
@@ -102,11 +102,13 @@ void loadScene(const char* _scenePath, bool _setPositionFromScene) {
 
     bool setPositionFromCurrentView = bool(m_scene);
 
-    auto scene = std::make_shared<Scene>();
+    auto scene = std::make_shared<Scene>(m_scene->path(), m_scene->userDefines());
```

Comment:
To behave consistently, we'd need to set the "path" of the new `Scene` object to `_scenePath` here, otherwise the scene will become very confused about which file it is supposed to use if you load another scene file after initialization. 

More generally though, it doesn't seem super useful for the scene to store this path in the first place.